### PR TITLE
Change font generation process (remove JS dependency)

### PR DESCRIPTION
**Pull request by Monceber**
Part 1 of the changes for improvement of handling fonts in the Verovio (support for more fonts without the need of re-compilation/re-generation).

This part addresses improvements to the font generation scripts, namely the following:
- removing PhantomJS dependency and replacing it with Python code
- simplifying parts of the generation scripts and removing parts of the redundant code
- added script to generate XML files for plain text fonts (previous added and reverted as part of #1691)

Changes that were done overall
-
- changed **generate_all.sh** script by removing dependency on PhantomJS as well generate-bbox.js script that was called with it
- combined two steps from **generate_all.sh** (extract glyphs and generate bboxes) into one steps, that extracts all glyphs
- glyphs are no longer extracted from font.SVG into **tmp/** folder - they are read in the **extract-bounding-boxes.py** and extracted into glyph.XML files
- added two dependencies to the Python code: `lxml` as direct replacement for the ElementTree (to allow ordering of attributes in the XML files and support of pretty print feature for readability); `svgpathtools` as tool to replace funcitonality done in **Js** part of the code (converting `@path` attribute from SVG file into bounding box to get coordinates, height, width, etc. about glyph)
- font.XML that is generated as result is pretty much the same as before, save for the addition of `@n` attribute which contains name of the glyph. As result, glyph.XML file that are generated in the **Font/** folder are generated without the glyphname.
- font files for all fonts are re-generated as well all glyphs for all fonts (glyphs are basically the same, but with different name)
- changed **generate_woff.sh** script - removed PhantomJS and long overdue Saxon9ee dependencies. Changed script to use plain font generation for the VerovioText.

Files added and removed in this PR
-
- added [fonts/extract-bounding-boxes.py](https://github.com/rism-digital/verovio/commit/0cc7805c7fa120b98dc30ac89504760857791e04#diff-0c586ae72ebb06284150493599ca55d0f48b7ccff3308975424b92699addc183)
- added [fonts/generate_plain_text.py](https://github.com/rism-digital/verovio/commit/13ce9bcbcdbcd88dbd7e3aef66b17f4d4a834cb8#diff-a134973c399ebba5be8bed2f52883bc1c311a6bb0e6e8048cd6c5d16e7347f7b)
- added [fonts/generate_text_font.sh](https://github.com/rism-digital/verovio/commit/13ce9bcbcdbcd88dbd7e3aef66b17f4d4a834cb8#diff-10c78a1e1516fc77a8aa96709000bd3ab7dcdab0470cbaca42f5aa31f0ce1a64)
- removed [fonts/generate-bbox.js](https://github.com/rism-digital/verovio/commit/7f24be125ffb6588cd187040c8f2df57b9d32734#diff-3456069055bbd260223fd0d11bdb00e8e030af6d3274bc9462549b93540253bf)
- removed [fonts/extract-glyphs.py](https://github.com/rism-digital/verovio/commit/7f24be125ffb6588cd187040c8f2df57b9d32734#diff-617d766d7e4051330d81fc4a20db7269c48c513e19c444b923eceac1be6bcb0e)
- XML files for glyphs for all fonts (basically renaming from GlyphCode-GlyphName.XML to GlyphCode.XML)

### CODE REVIEW

**Review by lpugin — CHANGES_REQUESTED** [review]
Looks good. Would it be possible to have the PR without the files re-generated? Having 3,000+ files changes makes it difficult to understand. We can always have another PR with the files re-generated after. Thanks

File: fonts/generate-smufl-code.py [hunk 1]
```diff
@@ -2,10 +2,11 @@
 
 import os
 import xml.etree.ElementTree as ET
+from typing import Dict
 
 # Define file output paths
-smufl_h_path = '../include/vrv/smufl.h'
-smufl_cpp_path = '../src/smufl.cpp'
+SMUFL_HEADER = "../include/vrv/smufl.h"
+SMUFL_SOURCE = '../src/smufl.cpp'
```
Comment (ahankinson) [inline fonts/generate-smufl-code.py:9]: consistent quotation marks

File: fonts/generate-smufl-code.py [hunk 2]
```diff
@@ -26,62 +27,58 @@ def get_file_content(filepath):
...
+def get_supported_glyph_codes() -> Dict:
+    """Retrieve dictionary with supported SMuFL codepoints and name."""
+    supported_xsl = ET.parse("supported.xsl")
+    glyphs = supported_xsl.findall(".//glyph")
+    supported_glyphs = {}
     for glyph in glyphs:
-        result[glyph.attrib["glyph-code"]] = glyph.attrib["smufl-name"]
-    return result
+        supported_glyphs[glyph.attrib["glyph-code"]
+                         ] = glyph.attrib["smufl-name"]
```
Comment (ahankinson) [inline fonts/generate-smufl-code.py:48]: This is a strange line break, before the closing square bracket...

**Comment by ahankinson** [comment]
`svgpathtools` requires `numpy` and `scipy` which are pretty big dependencies... 

https://github.com/mathandy/svgpathtools/blob/master/setup.py#L33

File: fonts/extract-bounding-boxes.py [hunk 3]
```diff
@@ -0,0 +1,256 @@
...
+import sys
+from typing import Dict
+# We can 1-to-1 replace ElementTree with lxml to support such features like
+# pretty_print and retaining order of attributes during the output
+# import xml.etree.ElementTree as ET
+from lxml import etree as ET
+from svgpathtools import Path
+
+# Define svg namespace
+svg_ns = "http://www.w3.org/2000/svg"
```
Comment (ahankinson) [inline fonts/extract-bounding-boxes.py:14]:
```suggestion
SVG_NS = "http://www.w3.org/2000/svg"
```

File: fonts/generate-smufl-code.py [hunk 4]
```diff
@@ -2,10 +2,11 @@
 
 import os
 import xml.etree.ElementTree as ET
+from typing import Dict
```
Comment (ahankinson) [inline fonts/generate-smufl-code.py:5]: PEP585 deprecated these type hints as of Python 3.9; is there a minimum Python version requirement?

File: fonts/generate-smufl-code.py (hunk 2 above)
Comment (rettinghaus) [reply fonts/generate-smufl-code.py:47]: This comes from autopep-8

**Review by lpugin — APPROVED** [review]

**Pull request merged**